### DISCUSSION TITLE
Failed when using FastAPI (uvicorn) worker

### DISCUSSION THREAD

**Discussion by juanrenos** [fastapi/fastapi#6164] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
cd .. && uvicorn --host=0.0.0.0 --port=8000 --root-path $ROOT_PATH logistics.app.main:app --workers 5 --timeout-keep-alive 900
```

### Description

Failed to deploy on the pipeline when I added --workers 5 at the uvicorn start. It works locally, but when in deployment it failed and returned Stopped reason Task failed container health checks (after taking so long time pipeline).  Without --workers 5 it runs properly.

Why I'm using --workers 5?, is because I'm using anyio task group, before --workers 5 sets, it returns Upstream request timeout. In another service --workers are working properly, but in this case, it is not.

### Operating System

Linux, Windows

### FastAPI Version

0.67.0

### Python Version

3.10

**Comment by JarroVGIT** [comment]
> Why I'm using --workers 5?, is because I'm using anyio task group, before --workers 5 sets, it returns Upstream request timeout.

I don’t understand this. 5 workers mean 5 processes, that has nothing to do with AnyIO task groups? Does it work with any N larger than 1?

**Comment by juanrenos** [comment]
> > Why I'm using --workers 5?, is because I'm using anyio task group, before --workers 5 sets, it returns Upstream request timeout.
> 
> I don’t understand this. 5 workers mean 5 processes, that has nothing to do with AnyIO task groups? Does it work with any N larger than 1?

I don't know if it's related or not, but in my previous work, I called 4 requests at the same time with anyio task group, and it returns "Upstream request timeout", after add --workers it does not return the "Upstream request timeout" again. 

And then I think --workers and my problem is related. And now I try to put --workers at logistic service, and it doesn't work in the pipeline (taking so long to deploy and then failing).

**Comment by tiangolo** [comment]
Thanks for the help @JarroVGIT ! ☕

I imagine you solved your problem, so thanks for closing the issue @juanrenos ! 😊
